### ISSUE TITLE
Adding/updating menus in Ribbon

### ISSUE THREAD

**Issue by rompik** [eryar/PipeCAD#73]
Hello Shing,
Are there any customization of Ribbon menu? Is it possible to add possibility to build ribbon menu by loading from xml file like in E3D?

**Comment by eryar** [comment]
The Ribbon menu use the 3rd lib SARibbon:
https://github.com/czyt1988/SARibbon

I found the lastest version SARibbon support customize by *.xml file.

I will check it.

**Comment by rompik** [comment]
Thank you! May be will be reasonable to create uic folder inside PipeCAD and upload xml files, also to Github. So if developed will create new Python functionality he will be able to add to menu and share with others w/out downloading new PipeCad distributive.

**Comment by eryar** [comment]
Yes, Let user to comtomize menu and link Python function themselves.

**Comment by eryar** [comment]
1. Define user actions in PipeCAD.Design.uic if you want to customize actions in Design Module;
![image](https://user-images.githubusercontent.com/8439181/178488323-0cae2f84-e0eb-4ea0-bc89-4d40d11ff46e.png)

2. In PipeCAD right upper button to customize Ribbon Bar;
![image](https://user-images.githubusercontent.com/8439181/178488714-e8228fe1-6074-4a82-a948-eec14a6f4051.png)

This will save to PipeCAD.Design.xml file.

**Comment by eryar** [comment]
https://github.com/eryar/PipeCAD/releases/tag/1.0.20

**Comment by rompik** [comment]
Hello Shing,
Please, can you advise how to setup path to external icons in xml files? Currently, I can use only icons from PipeCAD resources like here: 
```xml
    <Action Key="show_revision_manager" Text="Revision Manager" Icon=":/PipeCad/Resources/project_import.png" ToolTip="Revison Manager utiltiy" Module="omp.revmanager" Function="omp.revmanager.showRevManager()" />
```

**Comment by eryar** [comment]
Hello Roman,


The icon path is the path of QAction,
You can use a local path of the icon. 





在 2022-08-30 16:23:44，"Roman PARYGIN" ***@***.***> 写道：


Hello Shing,
Please, can you advise how to setup path to external icons in xml files? Currently, I can use only icons from PipeCAD resources like here:

    <ActionKey="show_revision_manager"Text="Revision Manager"Icon=":/PipeCad/Resources/project_import.png"ToolTip="Revison Manager utiltiy"Module="omp.revmanager"Function="omp.revmanager.showRevManager()" />
